### PR TITLE
Add Python usage example

### PR DESCRIPTION
**Pull request by dzenanz**

### CODE REVIEW

File: examples/SimpleMontage.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+ScalarPixelType = itk.UC
+OriginalPixelType = itk.RGBPixel[itk.UC]
+ScalarImageType = itk.Image[ScalarPixelType,dimension]
+OriginalImageType = itk.Image[OriginalPixelType,dimension]
+
+oImages = [] # RGB images, for mosaic creation
+sImages = [] # grayscale images, for registration
+for t in range(stageTiles.LinearSize()):
+  origin = stageTiles.Tiles[t].Position # AttributeError: 'itkTileConfiguration2' object has no attribute 'Tiles'
```
Comment (dzenanz) [inline examples/SimpleMontage.py:54]: Public member ivars are not exposed in Python. @thewtex is this expected?

Comment (thewtex) [reply examples/SimpleMontage.py:54]: According [to the documentation](http://swig.org/Doc4.0/SWIGPlus.html#SWIGPlus_member_data), they may be exposed. What is the result of `dir(stageTiles)`?

Comment (dzenanz) [reply examples/SimpleMontage.py:54]: dir(stageTiles)
`['LinearIndexToNDIndex', 'LinearSize', 'Parse', 'TryParse', 'Write', '__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__swig_destroy__', '__weakref__', 'getNextNonCommentLine', 'nDIndexToLinearIndex', 'parseLine', 'this', 'thisown']`

Comment (thewtex) [reply examples/SimpleMontage.py:54]: A getter may be required.

**Comment by dzenanz** [comment]
This currently hangs in `montage.Update()`. Stack trace:

Main thread(`futures[waited].get();`):
```text
>	_MontagePython.pyd!itk::TileMontage<itk::Image<unsigned char,2>,float>::GenerateData() Line 667	C++
 	_MontagePython.pyd!itk::ProcessObject::UpdateOutputData(itk::DataObject * __formal) Line 1708	C++
 	_MontagePython.pyd!itk::DataObject::UpdateOutputData() Line 379	C++
 	_MontagePython.pyd!itk::DataObject::Update() Line 309	C++
 	_MontagePython.pyd!itk::ProcessObject::Update() Line 1288	C++
 	_ITKCommonPython.pyd!_wrap_itkProcessObject_Update(_object * __formal, _object * args) Line 21966	C++
 	[External Code]	
 	SimpleMontage.py!<module> Line 77	Python
```
Active thread (`PyGILStateEnsure() { m_GIL = PyGILState_Ensure(); }`):
```text
>	_ITKPyUtilsPython.pyd!`anonymous namespace'::PyGILStateEnsure::PyGILStateEnsure() Line 28	C++
 	_ITKPyUtilsPython.pyd!itk::PyCommand::PyExecute() Line 116	C++
 	_ITKPyUtilsPython.pyd!itk::PyCommand::Execute(itk::Object * __formal, const itk::EventObject & __formal) Line 92	C++
 	_MontagePython.pyd!itk::SubjectImplementation::InvokeEventRecursion<itk::Object>(const itk::EventObject & event, itk::Object * self, std::reverse_iterator<std::_List_iterator<std::_List_val<std::_List_simple_types<itk::Observer *>>>> & i) Line 233	C++
 	_MontagePython.pyd!itk::SubjectImplementation::InvokeEvent(const itk::EventObject & event, itk::Object * self) Line 191	C++
 	_MontagePython.pyd!itk::Object::InvokeEvent(const itk::EventObject & event) Line 550	C++
 	_MontagePython.pyd!itk::ProcessObject::UpdateProgress(float progress) Line 1150	C++
 	_MontagePython.pyd!itk::TileMontage<itk::Image<unsigned char,2>,float>::GenerateData::__l9::<lambda>() Line 686	C++
 	[External Code]	
 	_MontagePython.pyd!itk::ThreadPool::AddWork::__l3::<lambda>() Line 98	C++
 	[External Code]	
 	_MontagePython.pyd!itk::ThreadPool::ThreadExecute() Line 180	C++
```

**Comment by thewtex** [comment]
We should be releasing the GIL during `Update()`.

What are the differences with threading in this filters?

**Comment by dzenanz** [comment]
Commenting out `UpdateProgress` call resolves the hang, see last commit.

File: include/itkTileMontage.hxx [hunk 2]
```diff
@@ -681,7 +682,7 @@ TileMontage<TImageType, TCoordinate>::GenerateData()
           this->RegisterPair(referenceIndex, currentTile);
           ++m_FinishedPairs;
           // all registrations finished = 95% of total progress
-          this->UpdateProgress(m_FinishedPairs * 0.95 / m_NumberOfPairs);
+          //this->UpdateProgress(m_FinishedPairs * 0.95 / m_NumberOfPairs);
```
Comment (thewtex) [inline include/itkTileMontage.hxx:685]: Can we live without this progress update when there is only 5% left?

Comment (dzenanz) [reply include/itkTileMontage.hxx:685]: This goes from 0% to 95%, and it is in this loop where the bulk of the processing time is spent.

Comment (thewtex) [reply include/itkTileMontage.hxx:685]: Ah.

Does it help to move the progress update to line 668 and 696?

Comment (dzenanz) [reply include/itkTileMontage.hxx]: That resolves the hang!

File: examples/SimpleMontage.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+if len(sys.argv) < 4:
+  print(f"Usage: {sys.argv[0]} <directoryWithInputData> <outputDirectory> <outputFilename>")
+  sys.exit(1)
+
+inputPath = sys.argv[1]
+outputPath = sys.argv[2]
+outFile = sys.argv[3]
+
+if not inputPath.endswith('/') and not inputPath.endswith('\\'):
```
Comment (thewtex) [inline examples/SimpleMontage.py:34]: To improve path handling, we should use `pathlib`.

```
from pathlib import Path

input_path = Path(sys.argv[1])

stage_tiles.Parse(str(input_path / "TileConfiguration.txt"))
```

File: examples/SimpleMontage.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+itkConfig.LazyLoading = False
+import itk
+
+itk.auto_progress(2)
+
+if len(sys.argv) < 4:
+  print(f"Usage: {sys.argv[0]} <directoryWithInputData> <outputDirectory> <outputFilename>")
+  sys.exit(1)
+
+inputPath = sys.argv[1]
```
Comment (thewtex) [inline examples/SimpleMontage.py:30]: Please use snake_case for variables through, e.g. `input_path`.

File: examples/SimpleMontage.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+outFile = sys.argv[3]
+
+if not inputPath.endswith('/') and not inputPath.endswith('\\'):
+  inputPath += '/'
+
+if not outputPath.endswith('/') and not outputPath.endswith('\\'):
+  outputPath += '/'
+
+os.chdir(outputPath)
+outFile = os.path.abspath(outFile)
```
Comment (thewtex) [inline examples/SimpleMontage.py:41]: Once a `Path` instance, we can get this [with `out_file.resolve()`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.resolve).

File: examples/SimpleMontage.py [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+import sys
+import os
+import itkConfig
+itkConfig.LazyLoading = False
+import itk
+
+itk.auto_progress(2)
+
+if len(sys.argv) < 4:
+  print(f"Usage: {sys.argv[0]} <directoryWithInputData> <outputDirectory> <outputFilename>")
```
Comment (thewtex) [inline examples/SimpleMontage.py:27]: `InputData` -> `InputTiles`

**Review by thewtex — CHANGES_REQUESTED** [review]

**Review by imikejackson — APPROVED** [review]

**Review by thewtex — APPROVED** [review]

**Pull request merged**